flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/PeriodicMaterializationManager.java

```diff
@@ -132,8 +136,14 @@ public synchronized void close() {
     public void triggerMaterialization() {
         mailboxExecutor.execute(
                 () -> {
-                    Optional<MaterializationRunnable> materializationRunnableOptional =
-                            keyedStateBackend.initMaterialization();
+                    metrics.reportStartedMaterialization();
+                    Optional<MaterializationRunnable> materializationRunnableOptional;
+                    try {
+                        materializationRunnableOptional = keyedStateBackend.initMaterialization();
+                    } catch (Exception ex) {
+                        metrics.reportFailedMaterialization();
+                        throw ex;
+                    }
 
                     if (materializationRunnableOptional.isPresent()) {
```

Comment:
In the opposite case (optional not present), shouldn't completed materialization be reported? Or maybe empty materialization?
My reasoning is that with no state updates, the number of started materializations might be much higher than the sum of completed and failed.